scripts/deploy.sh

```diff
@@ -122,19 +123,28 @@ if [[ ! -f /etc/openstack_deploy/user_osa_secrets.yml ]] && [[ -f /etc/openstack
   mv /etc/openstack_deploy/user_secrets.yml /etc/openstack_deploy/user_osa_secrets.yml
 fi
 
+# set permissions and lay down overrides files
+chmod 0440 /etc/openstack_deploy/user_*_defaults.yml
+if [[ ! -f "$OA_OVERRIDES" ]]; then
+  cp "${RPCD_DIR}"/etc/openstack_deploy/user_osa_variables_overrides.yml $OA_OVERRIDES
+fi
+if [[ ! -f "$RPCD_OVERRIDES" ]]; then
+  cp "${RPCD_DIR}"/etc/openstack_deploy/user_rpco_variables_overrides.yml $RPCD_OVERRIDES
+fi
+
 # Apply host security hardening with openstack-ansible-security
 # The is applied as part of setup-hosts.yml
 if [[ "$DEPLOY_HARDENING" == "yes" ]]; then
```

Comment:
I think this if block should be changed to something along the lines of:

```
if [[ "$DEPLOY_HARDENING" != "yes" ]]; then
  echo -e 'DEPLOY_HARDENING should no longer be used. Please add the following line to /etc/openstack_deploy/user_osa_variables_overrides.yml to disable security hardening.\n\napply_security_hardening: false\n'
  exit 1
fi
```

With the way that the variables files are split between defaults and overrides there is not an appropriate place for this script to write the var. The overrides files are supposed to come with a guarantee that we will not modify them and defaults files are supposed to contain defaults. In addition, given that the default is now to deploy security hardening, rerunning the script and forgetting to set DEPLOY_HARDENING to no on a deployment where it was disabled is likely to be problematic.